### ISSUE TITLE
Boot from NVMe is broken with "upstream" u-boot and "upstream" kernel

### ISSUE THREAD

**Issue by misuzu** [starfive-tech/VisionFive2#72]
Here is my setup:

* u-boot from https://github.com/misuzu/u-boot/commits/visionfive2
It's https://github.com/u-boot/u-boot/commit/a169438411f9277cc689c14078151aa1d1caae3c but with https://github.com/misuzu/u-boot/commit/9c29cddf93ba6122d92f285d3d015944ffa08c70 for the u-boot to actually work, [USB patch](https://patchwork.ozlabs.org/project/uboot/cover/20230807085338.27119-1-minda.chen@starfivetech.com/) (also tried without it - same issue) and https://github.com/misuzu/u-boot/commit/355b187cda897494d7a1436b6897999f633a5fc9 to make it boot from NVMe.

* Linux kernel from https://github.com/starfive-tech/linux/commit/64a6c57ec372c44d0f25416c9614657e274fccff

* systemd-boot as bootloader

* [NixOS](https://github.com/misuzu/nixos-vf2/tree/0.4)

The problem:

I have my EFI partition on an SD card and the root ext4 filesystem on an NVMe drive. This setup works fine, the u-boot loads systemd-boot, it then loads the kernel, it does its thing, mounts root filesystem and starts systemd.

But, if I copy my EFI partition to an NVMe drive and eject the SD card or just run `pci enum` and then `boot` in u-boot, the Linux kernel won't be able to initialize NVMe.

tl;dr - initialize NVMe drive in the u-boot - kernel won't find it, don't initialize NVMe drive in the u-boot - everything fine.

Here are the logs:
[efi_on_sdcard.log](https://github.com/starfive-tech/VisionFive2/files/12294859/efi_on_sdcard.log) (working)

[efi_on_nvme.log](https://github.com/starfive-tech/VisionFive2/files/12294860/efi_on_nvme.log) (fails)

[sdcard-to-nvme_cut_diff.txt](https://github.com/starfive-tech/VisionFive2/files/12294864/sdcard-to-nvme_cut_diff.txt) (the difference between the two, minus time)

The OS image and the u-boot files for testing can be found [here](https://github.com/misuzu/nixos-vf2/releases/tag/0.4).

**Comment by misuzu** [comment]
Looks like the issue is gone with https://github.com/starfive-tech/linux/commit/67e8df01b875afd312a7d2ab77f56a62f39dd6d9 and https://github.com/misuzu/u-boot/commit/0850e3a5b9d02cc5d9c06dbd22e97a16c67285e8
